### DISCUSSION TITLE
Line breaks in data in front matter of markdown file

### DISCUSSION THREAD

**Discussion by chmgui** [11ty/eleventy#1573] [Q&A]
Hi All
Newbie here.  Does anybody know how to include line breaks in the data of a variable in the Front Matter of a markdown file?
I tried the Indent Code Block in this link https://www.11ty.dev/docs/languages/markdown/ and it didn't work.
Tried using a vertical | as per this link https://discourse.getgrav.org/t/multi-line-strings-in-frontmatter/1226/5  and it also didn't work.  
https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines

Thank you very much

**Comment by pdehaan** [comment]
This worked locally for me

```md
---
title: HOME
eleventyComputed:
  excerpt: |
    ## {{ title }} subheading

    This is some content.
    Linebreaks?

    Or multi-blank lines for a line break?
author: pdehaan
---

# {{ title }}

{{ excerpt }}
```

Which generated the following output:

```html
<h1>HOME</h1>
<h2>HOME subheading</h2>
<p>This is some content.
Linebreaks?</p>
<p>Or multi-blank lines for a line break?</p>
```

---

**UPDATE:** Pushed my demo to https://github.com/pdehaan/11ty-1573-frontmatter-multiline

**Reply by chmgui** [reply]
Thank you pdehaan.
Okay it is not working because I am using the variable in a Nunjucks njk file.  My apologies for not being clear.
How to get it to work in .njk file ....
node --version
v15.5.0
npm --version
7.3.0
$ npx @11ty/eleventy --version
0.11.1

UPDATE - found the solution -` {{ service.data.description | striptags(true) | escape | nl2br }}`
https://mozilla.github.io/nunjucks/templating.html#striptags-value-preserve_linebreaks

Thank you once again Peter!

**Reply by pdehaan** [reply]
Ah sorry, I misunderstood. I thought you were using a .md file w/ frontmatter.

If you're using Nunjucks/Liquid, you'll need to create a Markdown filter or paired tag to convert the frontmatter markdown string (since Nunjucks/Liquid won't run output through a markdown parser). I updated https://github.com/pdehaan/11ty-1573-frontmatter-multiline/blob/master/src/nunjucks.njk and added a "markdown" filter to the [.eleventy.js config file](https://github.com/pdehaan/11ty-1573-frontmatter-multiline/blob/ec9295452772bb641c66376ca6836f28826d4f33/.eleventy.js#L6). And if you want to do a paired shortcode instead (ie: `{% markdown %}{{ excerpt }}{% endmarkdown %}`), there should be an example in https://github.com/11ty/eleventy/issues/658.

But yeah, if you only want to handle plain text w/ line breaks without markdown formatting, I think your solution of `nl2br` is _way_ better. :+1:
